src/lib/storage/storage_manager.cpp

```diff
@@ -503,53 +505,20 @@ std::shared_ptr<Chunk> StorageManager::map_chunk_from_disk(const uint32_t chunk_
 
   // TODO: Remove unneccesary map on whole file
   const auto* map = reinterpret_cast<uint32_t*>(mmap(NULL, file_bytes, PROT_READ, MAP_PRIVATE, fd, off_t{0}));
+  _maps.emplace_back(map);
```

Comment:
maybe we should store here a reference to what is stored in this map? Like f.e. the filename could work for now but in the future we should use a better DS here

Reply:
We are also not supposed to use raw_pointers.
But I think this map is not in the scope of this PR.